sdk/schemaregistry/azure-data-schemaregistry-avro/src/main/java/com/azure/data/schemaregistry/avro/SchemaRegistryAvroAsyncDeserializer.java

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.data.schemaregistry.avro;
+
+import com.azure.data.schemaregistry.SerializationException;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Scheduler;
+import reactor.core.scheduler.Schedulers;
+
+import java.util.concurrent.Executors;
+
+/**
+ * Asynchronous registry-based deserializer implementation.
+ */
+public class SchemaRegistryAvroAsyncDeserializer {
+    private static final int DEFAULT_THREAD_POOL_SIZE = 8;
+
+    private final SchemaRegistryAvroDeserializer deserializer;
+    private final Scheduler scheduler;
+
+    /**
+     * Instantiates instance of async deserializer.
+     *
+     * @param deserializer synchronous internal deserializer implementation
+     */
+    SchemaRegistryAvroAsyncDeserializer(SchemaRegistryAvroDeserializer deserializer) {
+        this.deserializer = deserializer;
+        this.scheduler = Schedulers.fromExecutor(Executors.newFixedThreadPool(DEFAULT_THREAD_POOL_SIZE));
+    }
+
+    /**
+     * Async wrapper around synchronous deserialization method
+     * @param data bytes containing schema ID and encoded byte representation of object
+     * @return Mono wrapper around deserialized object
+     * @throws SerializationException if deserialization operation fails
+     */
+    public Mono<Object> deserializeAsync(byte[] data) throws SerializationException {
```

Comment:
Don't need `async` suffix since the class is called `SchemaRegistryAvroAsyncDeserializer`

Reply:
Ok.  I have removed sync and async suffixes for Serializer and Deserializer